### DISCUSSION TITLE
WebAPIで取得できる情報に、Organizationsの情報も追加してほしい

### DISCUSSION THREAD

**Discussion by cold-wisteria** [increments/qiita-discussions#729] [要望・アイデア・不具合]
現在Qiitaで、WebAPI(以下APIと書きます)で取得できる情報はこちらの通りです。が、この中にOrganizationsに関するAPIはありません。
https://qiita.com/api/v2/docs

企業・組織の中でQiitaの活動を盛り上げるのに、チャットツールやグループウェアに対してOrganizationsでの活動・情報を連携したい、というニーズは一定以上あるのでは、と考えております。
が、現時点では「RSSで、そのOrganizationの新着記事を受け取る」以外の自動化・機械化は、公式に認められている方法ではできない認識です。

例えば

- Organizationsの記事投稿数・いいね数を週次で社内チャットツールに連携して、日々いいねが増えていく様で盛り上がる
- Organizationのメンバーごとのいいね数の合計を日次で可視化し、日々切磋琢磨する
- OrganizationのメンバーのQiita IDを日次で取得し、社内チャットツールのIDとのマッピングDBの情報を更新する

などの仕組みづくりを機械的に行うことが、現在ではできません。

そのため、以下をはじめとした、Organizationsページで閲覧できる情報について、APIでの取得ができるように、Organization関連情報に関するGETのAPIの拡充についてご検討いただけますと幸いです。

- 記事投稿数・いいね数・フォロワー数
- 記事一覧(paginationで)
- メンバー一覧(paginationで)

**Comment by tanaka-asami415** [comment]
@cold-wisteria 
リクエストありがとうございます🙏
有料プラン等でのご提供になるかもしれませんが、将来的に検討したいと思っています！

**Comment by tanaka-asami415** [comment]
@cold-wisteria 
Organizationのデータを取得するAPIはまだご用意できていないのですが、下記内容は記事データを取得するAPI の `org` オプションを利用して取得できる事に気がつきましたため、ご報告します。
前回の返信の際にご案内が出来ず申し訳ありません🙇

- 記事毎の「いいね数」
- 記事一覧

### APIで特定のOrganization記事一覧の取得の仕方

- `https://qiita.com/api/v2/items` で `org` オプションを利用する
    - 例: `https://qiita.com/api/v2/items?page=1&per_page=20&query=org:qiita-inc`
- 新規投稿順にしたい場合は `sort:created`を指定する
    - 例 `https://qiita.com/api/v2/items?page=1&per_page=20&query=org:qiita-inc+sort:created`

よろしくお願いします！

**Comment by k-nzw** [comment]
私も組織内でOrganizationを運営しておりますが、特にOrganizationのフォロワー数とフォロワーの一覧をAPIで取得できるようになると大変助かります。
自組織の記事に関心がある方がどれくらいいらっしゃるか、定期的に計測したいためです。
ご検討よろしくお願いいたします。

**Reply by tsuzuki-takaaki** [reply]
@k-nzw

ご提案ありがとうございます！
WebAPI の拡充は引き続き検討しておりますので、いただいた内容は貴重なご意見として取り入れさせていただきます！

よろしくお願いいたします🙇‍♂️

**Comment by sh0n0** [comment]
Organization一覧を取得できるAPIが欲しいです🙇
(https://github.com/increments/qiita-discussions/discussions/729#discussioncomment-10517518 で書かれているクエリパラメータに指定するOrganizationのnameを一覧取得したいです)

参考までに、用途としては https://zenn-publication-techstack.vercel.app/ のようにOrganizationの情報の可視化や検索をできるアプリケーションを考えています。

**Reply by zumi0** [reply]
@sh0n0 
リクエストありがとうございます🙏 具体的なユースケースや参考URLもご提示いただき非常に助かります。

Organization関連のAPI拡充については引き続き検討しておりますので、今回いただいたご要望も重要なフィードバックとして取り入れさせていただきます。

よろしくお願いします！
